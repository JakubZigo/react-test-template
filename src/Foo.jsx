import React, { Component } from 'react'

class Foo extends Component {
  constructor(props) {
    super(props)
    this.state = {
      data: 'Initial data'
    }
  }

  updateState() {
    this.setState({data: 'Data updated'})
  }

  render() {
    return (
      <div className="foo">
        <button onClick={() => this.updateState()}>CLICK</button>
        <h4>{this.state.data}</h4>
      </div>
    )
  }
}

export default Foo
