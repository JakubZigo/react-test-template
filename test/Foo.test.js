import React from 'react'
import { shallow, mount, render } from 'enzyme'
import toJson from 'enzyme-to-json'

import Foo from '../src/Foo'

describe('Foo', () => {
  const wrapper = shallow(<Foo data="hello world"/>)

  it('Should render h4 with init status', () => {
    expect(wrapper.find('h4').length).toBe(1)
    expect(wrapper.find('h4').at(0).text()).toBe('HELLO WORLD')
  })

  it('should change status when button is clicked', () => {
    wrapper.find('button').simulate('click')
    expect(wrapper.find('h4').at(0).text()).toBe('hello world')
  })

  it('should change state.data when button is clicked', () => {
    wrapper.find('button').simulate('click')
    expect(wrapper.state('data')).toBe('hello world')
  })

  it('should render correctly', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  })
})
