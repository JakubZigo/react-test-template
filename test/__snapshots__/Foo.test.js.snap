// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Foo should render correctly 1`] = `
<div
  className="foo"
>
  <button
    onClick={[Function]}
  >
    CLICK
  </button>
  <h4>
    hello world
  </h4>
</div>
`;
